File: RedSheet.DbEntities/Models/ApplicationExceptionLog.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ApplicationExceptionLogs", Schema = "dbo")]
	[Serializable]
	public partial class ApplicationExceptionLog  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ApplicationExceptionLogId { get; set; } 

	public Nullable<int> ApplicationModuleId { get; set; } 

	public Nullable<int> ApplicationTimeZoneId { get; set; } 
 
	[Required] 
	[Column(TypeName = "date")]
	public System.DateTime ExceptionDate { get; set; } 
 
	[Required]
	public string ExceptionSource { get; set; } 
 
	[Required]
	public string ExceptionType { get; set; } 
 
	[Required]
	public string InnerException { get; set; } 
 
	[Required]
	public string Message { get; set; } 
 
	[MaxLength(10)]
	public string RequestMethod { get; set; } 
 
	[Required]
	public string StackTrace { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string Url { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int UserId { get; set; } 

	public ApplicationExceptionLog()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/application-exception-log.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class ApplicationExceptionLog {
    constructor(applicationExceptionLog?: ApplicationExceptionLog )  {
        let properties = [ "applicationExceptionLogId", "applicationModuleId", "applicationTimeZoneId", "exceptionDate", "exceptionSource", "exceptionType", "innerException", "message", "requestMethod", "stackTrace", "url", "userId",];
        for (let property of properties)
            if (applicationExceptionLog && applicationExceptionLog[property])
                this[property] = applicationExceptionLog[property];
    }
 
	applicationExceptionLogId : number =   0 ;
 
	applicationModuleId : number =   undefined;
 
	applicationTimeZoneId : number =   undefined;
 
    @required()
	exceptionDate : Date =   undefined;
 
    @required()
	exceptionSource : string =   undefined;
 
    @required()
	exceptionType : string =   undefined;
 
    @required()
	innerException : string =   undefined;
 
    @required()
	message : string =   undefined;
 
    @maxLength(10)
	requestMethod : string =   undefined;
 
    @required()
	stackTrace : string =   undefined;
 
    @required()
    @maxLength(200)
	url : string =   undefined;
 
    @range(1,2147483647)
	userId : number =   undefined;


}
